packages/viz/src/lib/layer/LayerInteractivity.ts

```diff
@@ -21,122 +18,94 @@ export class LayerInteractivity {
   private _layerGetStyleFn: () => Style;
   private _layerSetStyleFn: (style: Style) => Promise<void>;
 
+  private _layerOnFn: EventHandler;
+  private _layerOffFn: EventHandler;
+
   private _layer: StyledLayer;
 
-  private _setCursorOnHover: boolean;
+  private _layerEmitFn: (type: string, event?: unknown) => void;
+
+  private _clickPopupHandler?: InteractionHandler;
+  private _hoverPopupHandler?: InteractionHandler;
 
   constructor(
     layer: StyledLayer,
     layerGetStyleFn: () => Style,
     layerSetStyleFn: (style: Style) => Promise<void>,
+    layerEmitFn: (type: string, event?: unknown) => void,
+    layerOnFn: EventHandler,
+    layerOffFn: EventHandler,
     hoverStyle?: Style | string,
     clickStyle?: Style | string
```

Comment:
Picky comment: I feel like we could refactor this a bit into an options object, cause the number of params is now a bit high and some of them are optional